docs: add extend-middleware guide (#1584)
web-infra-dev/rsbuild: packages/document/docs/en/guide/basic/server.md

:::tip
Rsbuild applications cannot use Rspack's [devServer](https://rspack.dev/config/dev-server) config. You can configure the behavior of the server through Rsbuild's `dev` and `server` configs, see [Config Overview](/config/index) for details.
:::

## Extend middleware

> Rsbuild does not use any framework, and the `req` and `res` in the Rsbuild middleware are all native Node objects.

Reviewer: ```suggestion
Rsbuild server does not use any Node.js frameworks, and the `req` and `res` provided by Rsbuild middleware are both native Node.js objects.
```

This means that when you migrate from other server-side frameworks (such as Express), the original middleware may not necessarily be used directly in Rsbuild. For example, the `req.params`,`req.path`, `req.search`, `req.query` and other objects provided by Express cannot be accessed in the Rsbuild middleware.

> If you want the middleware you originally used to still be available in Rsbuild, you can use the following method to pass in the server-side application as middleware:

Reviewer: ```suggestion
If you need to use existing middleware in Rsbuild, this can be done by passing the server application as middleware as follows:
```
